src/back-end/lib/crud.ts

```diff
@@ -86,3 +87,37 @@ export function makeRouter<SupportedRequestBodies, SupportedResponseBodies, Crea
     return routes;
   };
 }
+
+interface BodyWithErrors {
+  permissions?: string[];
+  database?: string[];
+}
```

Comment:
I suggest moving this to shared/lib/index.ts so you can extend this interface in the shared validation error interfaces